util-images/network/netperfbenchmark/pkg/worker/worker.go

```diff
@@ -19,48 +19,67 @@ limitations under the License.
 package worker
 
 import (
-	"bufio"
+	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
-	"io"
 	"net/http"
-	"os/exec"
+	"os"
 	"strconv"
 	"time"
 
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog"
 )
 
-// Worker hold data required for facilitating measurement.
+// Worker holds data required for facilitating measurement.
 type Worker struct {
-	stopCh       chan (struct{})
-	parsedResult []float64
-	err          error
-	protocol     string
-	workerDelay  time.Duration
-	startTime    time.Time
+	resourceInterface dynamic.ResourceInterface
+	podName           string
+	work              work
+	stopCh            chan struct{}
 }
 
-// MetricResponse is the response sent back to controller after collecting measurement.
-type MetricResponse struct {
-	Result      []float64
-	WorkerDelay time.Duration
+type work struct {
+	resourceContent map[string]interface{}
+	workType        string
+	resourceSpecMap map[string]interface{}
```

Comment:
Actually - let's remove this one - and just keep "resourceSpec" field below (it contains the same information).

It means that you need to change `extractParameters` method to take `resourceSpec` as input, but it's actually fine (and probably even easier to reason about).

Reply:
Modified